cpp/src/column/column.cu

```diff
@@ -162,11 +164,86 @@ __global__ void copy_offset_bitmask(bitmask_type *__restrict__ destination,
   }
 }
 
+/**---------------------------------------------------------------------------*
+ * @brief Copies the bits starting at the specified offset from a source
+ * bitmask into the destination bitmask.
+ *
+ * Bit `i` in `destination` will be equal to bit `i + offset` from `source`.
+ *
+ * @param destination The mask to copy into
+ * @param source The mask to copy from
+ * @param bit_offset The offset into `source` from which to begin the copy
+ * @param number_of_bits The number of bits to copy
+ *---------------------------------------------------------------------------**/
+template <int BLOCK_SIZE>
+__global__ void copy_offset_nullmask(bitmask_type *__restrict__ destination,
+                                     bitmask_type const *__restrict__ source,
+                                     size_type bit_offset,
+                                     size_type number_of_mask_words) {
+  constexpr size_type warp_size{32};
+  size_type destination_word_index = threadIdx.x + blockIdx.x * blockDim.x;
+  size_type source_word_index = destination_word_index + word_index(bit_offset);
+
+  if (destination_word_index < number_of_mask_words) {
+    bitmask_type curr_word = source[source_word_index];
+    bitmask_type next_word = 0;
+    if (destination_word_index + 1 < number_of_mask_words) {
```

Comment:
I think this `if`can  be removed and make the above `if` instead be ` if (destination_word_index <= number_of_mask_words)` ?

Reply:
If `destination_word_index == number_of_mask_words` then accessing `source[source_word_index]` would be equivalent to accessing
`source[destination_word_index + word_index(bit_offset)]` which would be equivalent to accessing
`source[number_of_mask_words + word_index(bit_offset)]` which is an out of bounds access.

I used that if inside hoping that the compiler will convert it to a conditional move which won't be problematic. I think we do need to load current word and avoid loading next word at the edge case so I am not sure I can combine both the ifs together.